src/zenml/steps/utils.py

```diff
@@ -410,19 +410,26 @@ def Do(
         return_type: Type[Any] = spec.annotations.get("return", None)
         if return_type is not None:
             if isinstance(return_type, Output):
-                # Resolve named (and multi-) outputs.
```

Comment:
Are you sure we can jut remove this block of code? What is the reasoning for removing this entire logic? Seems unrelated to the error message

Reply:
I think all the logic should still be the same, I just unified it into one for-loop instead of two so we don't have to duplicate everything

Reply:
OK had to zoom out into the file to see whats up. Much cleaner now, ill change my review!